src/cljs/org/broadinstitute/firecloud_ui/page/library/library_page.cljs

```diff
@@ -130,15 +130,47 @@
                   (callback {:error status-text})))})))))})
 
 (react/defc SearchSection
-  {:render
+  {:get-filters
    (fn [{:keys [props]}]
+     (utils/map-kv
+       (fn [k v]
+         [(name k) v])
+       (:facet-filters props)))
+   :render
+   (fn [{:keys [props this]}]
      [:div {}
       [:div {:style {:fontWeight 700 :fontSize "125%" :marginBottom "1em"}} "Search Filters:"]
       [:div {:style {:background (:background-light style/colors) :padding "16px 12px"}}
-       [comps/TextFilter {:ref "text-filter"
-                          :initial-text (:search-text props)
-                          :width "100%" :placeholder "Search"
-                          :on-filter (:on-filter props)}]]])})
+       [comps/AutocompleteFilter
+        {:on-filter (:on-filter props)
+         :width "100%"
+         :initial-text (:search-text props)
+         :ref "text-filter"
+         :placeholder "Search"
+         :facet-filters (:facet-filters props)
+         :bloodhoundInfo {:url (str (config/api-url-root) "/api/library/suggest")
+                          :transform (fn [response]
+                                       (clj->js
+                                         (mapv
+                                           (fn [string]
+                                             {:value string})
+                                           (aget response "results"))))
```

Comment:
Not sure which is better, but `(fn [string] {:value string})` could be `#(hash-map :value %)` or `(partial hash-map :value)`.

I kinda like the look of:

```clojure
(mapv (partial hash-map :value)
      (aget response "results"))
```

But maybe that's a bit harder to read?  It just makes me feel a little warm and fuzzy when you can express something without repeating identifiers.